File: ios/TennisAce/Models/Surface.swift
import Foundation

/// Court surface. `rawValue` is the stable wire/storage contract
/// (see docs/DOMAIN_MODEL.md); `displayName` is for the UI.
enum Surface: String, CaseIterable, Hashable {
    case hard
    case clay
    case grass
    case carpet

    var displayName: String {
        switch self {
        case .hard: "Hartplatz"
        case .clay: "Sand"
        case .grass: "Rasen"
        case .carpet: "Teppich"
        }
    }

    static func from(_ value: String) -> Surface {
        Surface(rawValue: value) ?? .hard
    }
}

File: android/app/src/main/java/investments/hoelzl/tennisace/model/Surface.kt
package investments.hoelzl.tennisace.model

/**
 * Court surface. [value] is the stable wire/storage contract (see
 * docs/DOMAIN_MODEL.md); [displayName] is for the UI.
 */
enum class Surface(val value: String, val displayName: String) {
    HARD("hard", "Hartplatz"),
    CLAY("clay", "Sand"),
    GRASS("grass", "Rasen"),
    CARPET("carpet", "Teppich");

    companion object {
        fun fromValue(value: String): Surface =
            entries.firstOrNull { it.value == value } ?: HARD
    }
}
